src/XMLInterface.cpp

```diff
@@ -518,4 +520,237 @@ void XMLInterface::writeXML(Scores& fullset, ProteinProbEstimator* protEstimator
+  } else {
+    time +=  hour_s;
+  }
+
+   int min = ltm->tm_min;
+
+   std::string min_s = to_string(min);
+   if (min_s.length() < 2) {
+    time += ":0" + min_s;
+  } else {
+    time += ":" + min_s;
+  }
+
+   int sec = ltm->tm_sec;
+   std::string sec_s = to_string(sec);
+   if (sec_s.length() < 2) {
+    time += ":0" + sec_s;
+  } else {
+    time += ":" + sec_s;
+  }
+
+   atomic_date = date + "T" + time;
+} */
+
+/* std::string getAtomicTime() {return atomicDate} */
+
+
+void XMLInterface::writePepXML(Scores& fullset, ProteinProbEstimator* protEstimator, std::string call) {
+
+  /* setAtomicTime(); */
+
+  ofstream os;
+  const string schema = // space +
+      "http://sashimi.sourceforge.net/schema_revision/pepXML/pepXML_v122.xsd";
+
+
+  os.open(pepXMLOutputFN_.data(), ios::out | ios::binary);
+  os << "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" << endl;
+  os << "<msms_pipeline_analysis date=\""<< getAtomicTime() <<"\" xmlns=\"http://regis-web.systemsbiology.net/pepXML\" summary_xml=\"/sdd/proteomics/DATA/PXD014076/iProphet3/interact-Symb_Proteome_DIA_RAW_S05_Q3.pep.xml\" xsi:schemaLocation=\"http://regis-web.systemsbiology.net/pepXML\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\">";
+  os << "<analysis_summary analysis=\"peptideprophet\" time=\"" << getAtomicTime() << "\">";
+  os << "</analysis_summary>";
+  
+
+  
+  std::string pepPath = pepXMLOutputFN_.data();
+
+  /* TODO: MUST FIX */
+  std::string base_name = pepPath.substr(0, pepPath.find("."));
+
+  ifstream ifs_psms(xmlpeptideOutputFN_PSMs.data(), ios::in | ios::binary);
+  os << ifs_psms.rdbuf();
+  ifs_psms.close();
+  remove(xmlpeptideOutputFN_PSMs.c_str());
+
+
+  os << "    </msms_run_summary>" << endl;
+  
+  os << "</msms_pipeline_analysis>" << endl;
+  os.close();
+}
+
+/* Get time-stamp for pepXML */
+void XMLInterface::setAtomicTime() {
+  time_t now = time(0);
+   
+   // convert now to string form
+   tm *ltm = localtime(&now);
+
+   std::string date = "";
+   std::string time = "";
+
+
+   int year = ltm->tm_year;
+   date += to_string(1900 + year);
+
+   int month = 1 + ltm->tm_mon;
+   std::string month_s = to_string(month);
+   
+   if (month_s.length() < 2) {
+     date +=  "-0" + month_s;
+   } else {
+     date +=  "-" + month_s;
+   }
+
+  int day = ltm->tm_mday;
+  std::string day_s = to_string(day);
+
+  if (day_s.length() < 2) {
+    date += "-0" + day_s;
+  } else {
+    date += "-" + day_s;
+  }
+
+   int hour = 5 + ltm->tm_hour;
+   std::string hour_s = to_string(hour);
+   if (hour_s.length() < 2) {
+    time += "0" + hour_s;
+  } else {
+    time +=  hour_s;
+  }
+
+   int min = ltm->tm_min;
+
+   std::string min_s = to_string(min);
+   if (min_s.length() < 2) {
+    time += ":0" + min_s;
+  } else {
+    time += ":" + min_s;
+  }
+
+   int sec = ltm->tm_sec;
+   std::string sec_s = to_string(sec);
+   if (sec_s.length() < 2) {
+    time += ":0" + sec_s;
+  } else {
+    time += ":" + sec_s;
+  }
+
+   atomicDate = date + "T" + time;
```

Comment:
Is it not better to pass this variable back to the caller, rather than saving it as a member?

Reply:
So I have to get the AtomicTime in two different functions i.e., writePepXML_PSMs and writePepXML, and the time has to be the same in both. So I have to create Set and Get function, i.e., setAtomicTime , and getAtomicTime.